### ISSUE TITLE
Function Request for Remote Comfy

### ISSUE THREAD

**Issue by aniketgore** [NimaNzrii/comfyui-photoshop#24]
Can you please make this work for remote comfy server?

**Comment by thenickdude** [comment]
The Photoshop and ComfyUI frontend is hardcoded to connect to 127.0.0.1. If you could enter a custom IP address and port then you could connect to a ComfyUI instance running on a remote machine instead.

**Comment by FugueSegue** [comment]
I create my artwork on a Wacam Cintiq using Photoshop. The Cintiq is not powerful enough to run Stable Diffusion software. So I purchased an Nvidia A5000 with 24GB VRAM, built a computer for it, and I only run SD software on it that I access via LAN (local area network).

I was able to install this plugin, custom node, and workflow with no problem. It functions if I run it on the same computer as Photoshop. But there is no way to connect it to Photoshop across a LAN. This is vital. Many professionals have opted for the same network arrangement as mine. It is better to dedicate a computer to SD software and access it across a LAN instead of running it on the same computer as other programs such as Photoshop. Most people who use Photoshop do not have high-VRAM cards. Accessing SD remotely is vital for many users.

**Comment by caniyabanci76** [comment]
+1 for this.  I normally have the ComfyUI server running on a different machine on the same local network.

**Comment by Visual-Synthesizer** [comment]
Ive got similar setup. having fun building my workflows, thanks. it works on windows PS workstation machine with two 3090 GPU's. Prints this at end and works great:

```
To see the GUI go to: http://127.0.0.1:8192
FETCH DATA from: O:\*******\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\extension-node-map.json
_PS_ Photoshop launched('127.0.0.1', 61712)
_PS_ comfyUi Not Connected
_PS_ Photoshop node added('127.0.0.1', 61713)
_PS_ Photoshop node added('127.0.0.1', 61716)

```

When try to run on my LAN network to my arch linux server with 4 A6000 GPU's it wont connect to the PS plugin, "i get couldn't load" in PS and does not print the final output in terminal. I run VS-CODE on the windows machine to with SSH key exchange all setup to port forward. Not sure if this is helpful, but if i manually press "Queue" on the comfyUI backend i get this error:



```
Using optimized recursive execution.
First run patching recursive_output_delete_if_changed and recursive_will_execute.
Note: If execution seems broken due to forward changes, you can disable the optimization from settings.

!!! Exception during processing!!! Failed to load directory after 5 attempts. 
 🔴 Make sure you have installed and started the Plugin Successfully. 
 🔴 Otherwise, you can restart your application and your plugin to fix this problem.

Traceback (most recent call last):
  File "node-Plugin.py", line 70, in loadConfig
    with open(self.configJson, "r") as file:
FileNotFoundError: [Errno 2] No such file or directory: 'None/config.json'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "node-Plugin.py", line 70, in loadConfig
    with open(self.configJson, "r") as file:
FileNotFoundError: [Errno 2] No such file or directory: 'None/config.json'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "node-Plugin.py", line 70, in loadConfig
    with open(self.settings, "r") as file:
FileNotFoundError: [Errno 2] No such file or directory: 'None/config.json'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "execution.py", line 151, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "node-Plugin.py", line 33, in execute
    self.loadConfig()
  File "node-Plugin.py", line 75, in loadConfig
    self.loadConfig(retry_count + 1)
  File "node-Plugin.py", line 75, in loadConfig
    self.loadConfig(retry_count + 1)
  File "node-Plugin.py", line 75, in loadConfig
    self.loadConfig(retry_count + 1)
  [Previous line repeated 1 more time]
  File "node-Plugin.py", line 77, in loadConfig
    raise Exception("Failed to load directory after 5 attempts. \n 🔴 Make sure you have installed and started the Plugin Successfully. \n 🔴 Otherwise, you can restart your application and your plugin to fix this problem")
Exception: Failed to load directory after 5 attempts. 
 🔴 Make sure you have installed and started the Plugin Successfully. 
 🔴 Otherwise, you can restart your application and your plugin to fix this problem

Prompt executed in 2.51 seconds

```

I even tried port forwarding the port that the windows machine was opening with the plugin but didnt help.

**Comment by aniketgore** [comment]
This all. I want to connect the plugin to my remote comfy installation on 192...:8080.

**Comment by NimaNzrii** [comment]
Coding it for you❤️
Currently I'm adding advance setting button to the plugin

Which contains custom ipadress as well

**Comment by aniketgore** [comment]
Awesome. Thanks.

Thanks
Aniket

**Comment by NimaNzrii** [comment]
> Ive got similar setup. having fun building my workflows, thanks. it works on windows PS workstation machine with two 3090 GPU's. Prints this at end and works great:
> 
> ```
> To see the GUI go to: http://127.0.0.1:8192
> FETCH DATA from: O:\*******\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\extension-node-map.json
> _PS_ Photoshop launched('127.0.0.1', 61712)
> _PS_ comfyUi Not Connected
> _PS_ Photoshop node added('127.0.0.1', 61713)
> _PS_ Photoshop node added('127.0.0.1', 61716)
> ```
> 
> When try to run on my LAN network to my arch linux server with 4 A6000 GPU's it wont connect to the PS plugin, "i get couldn't load" in PS and does not print the final output in terminal like it did in windows. I run VS-CODE on the windows machine to with SSH key exchange all setup to port forward. Not sure if this is helpful, but if i manually press "Queue" on the comfyUI backend i get this error:
> 
> ```
> Using optimized recursive execution.
> First run patching recursive_output_delete_if_changed and recursive_will_execute.
> Note: If execution seems broken due to forward changes, you can disable the optimization from settings.
> 
> !!! Exception during processing!!! Failed to load directory after 5 attempts. 
>  🔴 Make sure you have installed and started the Plugin Successfully. 
>  🔴 Otherwise, you can restart your application and your plugin to fix this problem.
> 
> Traceback (most recent call last):
>   File "node-Plugin.py", line 70, in loadConfig
>     with open(self.configJson, "r") as file:
> FileNotFoundError: [Errno 2] No such file or directory: 'None/config.json'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "node-Plugin.py", line 70, in loadConfig
>     with open(self.configJson, "r") as file:
> FileNotFoundError: [Errno 2] No such file or directory: 'None/config.json'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "node-Plugin.py", line 70, in loadConfig
>     with open(self.settings, "r") as file:
> FileNotFoundError: [Errno 2] No such file or directory: 'None/config.json'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "execution.py", line 151, in recursive_execute
>     output_data, output_ui = get_output_data(obj, input_data_all)
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "node-Plugin.py", line 33, in execute
>     self.loadConfig()
>   File "node-Plugin.py", line 75, in loadConfig
>     self.loadConfig(retry_count + 1)
>   File "node-Plugin.py", line 75, in loadConfig
>     self.loadConfig(retry_count + 1)
>   File "node-Plugin.py", line 75, in loadConfig
>     self.loadConfig(retry_count + 1)
>   [Previous line repeated 1 more time]
>   File "node-Plugin.py", line 77, in loadConfig
>     raise Exception("Failed to load directory after 5 attempts. \n 🔴 Make sure you have installed and started the Plugin Successfully. \n 🔴 Otherwise, you can restart your application and your plugin to fix this problem")
> Exception: Failed to load directory after 5 attempts. 
>  🔴 Make sure you have installed and started the Plugin Successfully. 
>  🔴 Otherwise, you can restart your application and your plugin to fix this problem
> 
> Prompt executed in 2.51 seconds
> ```
> 
> I even tried port forwarding the port that the windows machine was opening for the PS plugin but this didnt help.

Yo can forward the port but it's not the onlything that should be changed, we need to change the whole saving option in plugin, so it's not that easy

**Comment by joetcurrie** [comment]
Has there been any developments on this? I am also using comfyui with remote connection to PC, happy to test the plugin if that helps.

**Comment by NimaNzrii** [comment]
> Ive got similar setup. having fun building my workflows, thanks. it works on windows PS workstation machine with two 3090 GPU's. Prints this at end and works great:
> 
> ```
> To see the GUI go to: http://127.0.0.1:8192
> FETCH DATA from: O:\*******\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Manager\extension-node-map.json
> _PS_ Photoshop launched('127.0.0.1', 61712)
> _PS_ comfyUi Not Connected
> _PS_ Photoshop node added('127.0.0.1', 61713)
> _PS_ Photoshop node added('127.0.0.1', 61716)
> ```
> 
> When try to run on my LAN network to my arch linux server with 4 A6000 GPU's it wont connect to the PS plugin, "i get couldn't load" in PS and does not print the final output in terminal like it did in windows. I run VS-CODE on the windows machine to with SSH key exchange all setup to port forward. Not sure if this is helpful, but if i manually press "Queue" on the comfyUI backend i get this error:
> 
> ```
> Using optimized recursive execution.
> First run patching recursive_output_delete_if_changed and recursive_will_execute.
> Note: If execution seems broken due to forward changes, you can disable the optimization from settings.
> 
> !!! Exception during processing!!! Failed to load directory after 5 attempts. 
>  🔴 Make sure you have installed and started the Plugin Successfully. 
>  🔴 Otherwise, you can restart your application and your plugin to fix this problem.
> 
> Traceback (most recent call last):
>   File "node-Plugin.py", line 70, in loadConfig
>     with open(self.configJson, "r") as file:
> FileNotFoundError: [Errno 2] No such file or directory: 'None/config.json'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "node-Plugin.py", line 70, in loadConfig
>     with open(self.configJson, "r") as file:
> FileNotFoundError: [Errno 2] No such file or directory: 'None/config.json'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "node-Plugin.py", line 70, in loadConfig
>     with open(self.settings, "r") as file:
> FileNotFoundError: [Errno 2] No such file or directory: 'None/config.json'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "execution.py", line 151, in recursive_execute
>     output_data, output_ui = get_output_data(obj, input_data_all)
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "node-Plugin.py", line 33, in execute
>     self.loadConfig()
>   File "node-Plugin.py", line 75, in loadConfig
>     self.loadConfig(retry_count + 1)
>   File "node-Plugin.py", line 75, in loadConfig
>     self.loadConfig(retry_count + 1)
>   File "node-Plugin.py", line 75, in loadConfig
>     self.loadConfig(retry_count + 1)
>   [Previous line repeated 1 more time]
>   File "node-Plugin.py", line 77, in loadConfig
>     raise Exception("Failed to load directory after 5 attempts. \n 🔴 Make sure you have installed and started the Plugin Successfully. \n 🔴 Otherwise, you can restart your application and your plugin to fix this problem")
> Exception: Failed to load directory after 5 attempts. 
>  🔴 Make sure you have installed and started the Plugin Successfully. 
>  🔴 Otherwise, you can restart your application and your plugin to fix this problem
> 
> Prompt executed in 2.51 seconds
> ```
> 
> I even tried port forwarding the port that the windows machine was opening for the PS plugin but this didnt help.

# Fixed in V1.5.0 ✨
this problem should be fixed in ver 1.5.0
please update plugin and the node and let me know if it helped

**Comment by lonrencn** [comment]
···
# PS: client-tjxu62fly Photoshop Connected
# PS: PS not connected
[ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/extension-node-map.json
got prompt
model weight dtype torch.float16, manual cast: None
model_type EPS
Using xformers attention in VAE
Using xformers attention in VAE
VAE load device: cuda:0, offload device: cpu, dtype: torch.float32
CLIP/text encoder model load device: cuda:0, offload device: cpu, current: cpu, dtype: torch.float16
model weight dtype torch.float16, manual cast: None
model_type EPS
Using xformers attention in VAE
Using xformers attention in VAE
VAE load device: cuda:0, offload device: cpu, dtype: torch.float32
CLIP/text encoder model load device: cuda:0, offload device: cpu, current: cpu, dtype: torch.float16
Requested to load SD1ClipModel
loaded completely 9.5367431640625e+25 235.84423828125 True
Requested to load AutoencoderKL
loaded completely 9.5367431640625e+25 319.11416244506836 True
Requested to load BaseModel
loaded completely 9.5367431640625e+25 1639.4336013793945 True
100%|██████████████████████████████████████████████████████████████████████████████████████████| 24/24 [00:03<00:00,  7.24it/s]
Requested to load AutoencoderKL
loaded completely 9.5367431640625e+25 319.11416244506836 True
_PS_ error on send2Ps: Cannot connect to host 127.0.0.1:8188 ssl:default [Connect call failed ('127.0.0.1', 8188)]
Prompt executed in 6.86 seconds
